packages/vite/src/node/server/index.ts

```diff
@@ -489,10 +487,8 @@ export async function createServer(
   middlewares.use(serveRawFsMiddleware(server))
   middlewares.use(serveStaticMiddleware(root, server))
 
-  const isMiddlewareMode = middlewareMode && middlewareMode !== 'html'
-
   // spa fallback
-  if (config.spa && !isMiddlewareMode) {
+  if (config.appType === 'spa' && !middlewareMode) {
```

Comment:
IIUC, if `middlewareMode: true`, there would never be spa fallback. Is this correct?

I think if we run in middleware mode, we should still handle spa when `appType: 'spa'`, according to the docs. So we should remove `&& !middlewareMode` here.

Reply:
Yes, you are right. This is a good sign of the simplification that this option brings, and of me forgetting to implement half of the PR 😬